Source/WTF/wtf/FileSystem.cpp

```diff
@@ -833,6 +833,30 @@ String pathByAppendingComponents(StringView path, const Vector<StringView>& comp
 
 #endif
 
+#if !OS(WINDOWS) && !PLATFORM(COCOA)
+
+String createTemporaryDirectory()
+{
+    std::error_code ec;
+    std::string tempDir = std::filesystem::temp_directory_path(ec);
+    if (ec)
+        return String();
+
+    std::string newTempDirTemplate = tempDir;
+    newTempDirTemplate += "XXXXXXXX";
+
+    Vector<char> newTempDir(newTempDirTemplate.c_str(), newTempDirTemplate.size());
+
```

Comment:
nit: I don't think we need this new line.

Reply:
I've made the changes and I think it's ready